### PR TITLE
Add disk SMART data to collect-info tarball

### PR DESCRIPTION
**Pull request by andrewd-zededa**
SMART_details.json and SMART_details_previous.json

### CODE REVIEW

**Comment by andrewd-zededa** [comment]
@eriknordmark Should this go into any of the -stable branches?

File: pkg/debug/scripts/collect-info.sh [hunk 1]
```diff
@@ -373,6 +373,7 @@ ln -s /persist/netdump      "$DIR/persist-netdump"
 ln -s /persist/kcrashes     "$DIR/persist-kcrashes"
 ln -s /run                  "$DIR/root-run"
 cp -r /sys/fs/cgroup/memory "$DIR/sys-fs-cgroup-memory" >/dev/null 2>&1
+ln -s /persist/SMART_details* "$DIR/"
```
Comment (rouming) [inline pkg/debug/scripts/collect-info.sh:376]: If files are missing you end up having this link:
```
ln -s tmp/SMART_details* tmp/tmp/
ls -la tmp/tmp/
'SMART_details*' -> 'tmp/SMART_details*'
```
Are those jsons always there?

File: pkg/debug/scripts/collect-info.sh [hunk 2]
```diff
@@ -373,7 +373,7 @@ ln -s /persist/netdump      "$DIR/persist-netdump"
 ln -s /persist/kcrashes     "$DIR/persist-kcrashes"
 ln -s /run                  "$DIR/root-run"
 cp -r /sys/fs/cgroup/memory "$DIR/sys-fs-cgroup-memory" >/dev/null 2>&1
-ln -s /persist/SMART_details* "$DIR/"
+stat -t /persist/SMART_details.json > /dev/null 2>&1 && ln -s /persist/SMART_details* "$DIR/"
```
Comment (rouming) [inline pkg/debug/scripts/collect-info.sh:376]: Please `if [ -f /persist/SMART_details.json ] etc`, `stat` difficult to read .
Also please squash this commit to the previous one. We don't collect commits iteratively.

**Comment by andrewd-zededa** [comment]
@rouming shortened the file existence check and squashed down to one commit

**Review by rouming — APPROVED** [review]

**Pull request merged**